src/status_im2/common/bottom_sheet_screen/view.cljs

```diff
@@ -46,40 +43,54 @@
+                              (js/setTimeout #(reset! animating? false) ms))]
     (fn [{:keys [content skip-background? theme]}]
       (let [insets           (safe-area/get-insets)
             {:keys [height]} (rn/get-window)
-            padding-top      (:top insets)
-            padding-top      (if platform/ios? padding-top (+ padding-top 10))
             opacity          (reanimated/use-shared-value 0)
             translate-y      (reanimated/use-shared-value height)
             close            (fn []
+                               (set-animating-true)
                                (reanimated/animate translate-y height 300)
                                (reanimated/animate opacity 0 300)
-                               (rf/dispatch [:navigate-back]))]
+                               (rf/dispatch [:navigate-back]))
+            reset-open-sheet (fn []
+                               (reanimated/animate translate-y 0 300)
+                               (reanimated/animate opacity 1 300)
+                               (set-animating-false 300)
+                               (reset! scroll-enabled true))]
         (rn/use-effect
          (fn []
            (reanimated/animate translate-y 0 300)
-           (reanimated/animate opacity 1 300)))
+           (reanimated/animate opacity 1 300)
+           (set-animating-false 300)))
         (hooks/use-back-handler close)
-        [rn/view
-         {:style {:flex        1
-                  :padding-top padding-top}}
+        [rn/view {:style (style/container insets)}
          (when-not skip-background?
            [reanimated/view {:style (style/background opacity)}])
          [gesture/gesture-detector
-          {:gesture (drag-gesture translate-y opacity scroll-enabled curr-scroll close)}
+          {:gesture (drag-gesture {:translate-y        translate-y
+                                   :opacity            opacity
+                                   :scroll-enabled     scroll-enabled
+                                   :curr-scroll        curr-scroll
+                                   :close              close
+                                   :reset-open-sheet   reset-open-sheet
+                                   :set-animating-true set-animating-true})}
           [reanimated/view {:style (style/main-view translate-y theme)}
            [rn/view {:style style/handle-container}
             [rn/view {:style (style/handle theme)}]]
            [content
-            {:insets         insets
-             :close          close
-             :scroll-enabled scroll-enabled
-             :current-scroll curr-scroll
-             :on-scroll      #(on-scroll % curr-scroll)}]]]]))))
+            {:insets           insets
+             :close            close
+             :scroll-enabled   scroll-enabled
+             :current-scroll   curr-scroll
+             :on-scroll        #(on-scroll % curr-scroll)
+             :sheet-animating? animating?}]]]]))))
```

Comment:
Not a big fan of passing atoms as props to other components :cry:  but this atom will travel a long path, tested it and using a late-deref approach improved the performance a little :smile:

Reply:
hmm, have you considered using react context instead? it will help prevent the prop drilling, additionally  you can instead pass accessor functions rather than the ratom. wdyt? 👍

Reply:
It sounds as a good idea in terms of not  passing the props through many components.

But, I think re-frame might be a simpler option. TBH, I know they solve the same problem, it seems just a matter of  the JS world solution vs the Clojure world solution.

Right now, we are passing many atoms to the child of this component (the one I added is just another one). So we can consider adding either react-context or re-frame sub in a later refactoring to this component.

Also, we should consider if moving this helps in terms of readability and performance.

Thanks for the suggestion Jamie! Very interesting topic